js/validation.js

```diff
@@ -0,0 +1,154 @@
+    };
+
+    if (mail.length === 0) {
+        createBubble ('Please provide an e-mail adress');
+        return false;
+    } else if (whitespaceCheck(mail)) {
+        createBubble('Email can\'t have whitespaces');
+        return false;
+    } else if (simpleMailCheck(mail)) {
+        createBubble ('Email not correct');
+        return false;
+    }
+    return true;
+}
+
+// alternative solution to consider: pattern with regexp
+function validatePass(password) {        // checks if password has at least one letter, digit and special sign and is at least 8 characters long
+    const hasAnyLetter = function(stringToTest) {
+        return !(/[a-z]+/ig).test(stringToTest)
+    };
+    const hasAnyNumber = function(stringToTest) {
+        return !(/[\d]+/.test(stringToTest));
+    };
+    const hasAnySpecial = function(stringToTest) {
+        return !(/[!@#$%^&*()/\\}\[\].{,'";:|=+_\-?><~`]+/).test(stringToTest);
+    };
+
+    if (password.length < 8) {
+        createBubble ('Password must be at least 8 letters long');
+        return false;
+    } else if (hasAnyLetter(password)) {
+        createBubble ('Password must contain at least 1 letter');
+        return false;
+    } else if (hasAnyNumber(password)) {
+        createBubble ('Password must contain at least 1 digit');
+        return false;
+    } else if (hasAnySpecial(password)) {
+        createBubble ('Password must contain at least 1 special sign');
+        return false;
+    }
+    return true;
+}
+
+function validateVID(vidNumber) {       // checks if vid contains only between 1 and 5 digits and isn't empty
+    const checkVID = function(number) {
+        return !(/^[\d]{1,5}$/.test(number));
+    };
+
+    if (vidNumber.length === 0) {
+        createBubble('VID field cannot be left empty');
+        return false;
+    } else if (checkVID(vidNumber)) {
+        createBubble('VID number can only contain 1 to 5 digits');
+        return false;
+    }
+    return true;
+}
+
+function validateAmount(ticketsAmount) {        // checks if ticket amount field isn't empty, value is less than 20 and has only digits
+    const checkIfNumbers = function(number) {
+        return !(/^[\d]{1,2}$/.test(number));
+    };
+
+    if (ticketsAmount.length === 0) {
+        createBubble ('Fill in the ticket amount');
+        return false;
+    } else if (ticketsAmount > 20) {
+        createBubble ('You can only buy 20 tickets at once');
+        return false;
+    } else if (ticketsAmount < 1 || checkIfNumbers(ticketsAmount)) {
+        createBubble ('Incorrect ticket amount');
+        return false;
+    }
+    return true;
+}
+
+function saveToLocalStorage() {
+    const formArray = [
+        'first_name',
+        'last_name',
+        'textarea_1',
+        'textarea_2',
+        'email',
+        'password',
+        'vid_number',
+        'tickets_count'
+    ];
+    for (let x = 0; x < formArray.length; x++) {        // Gets every field from form and saves it as a pair field:value
```

Comment:
When you only want to walk through an array to do something on each element, Array.forEach is helpful :
```javascript
formArray.forEach(name => {
    localStorage.setItem(name, document.getElementsByName(name)[0].value);
});
```